### ISSUE TITLE
Legend order changes after toggling legend item

### ISSUE THREAD

**Issue by Pixcalibur** [c3js/c3#1945]
When the chart has hidden columns when initialised, toggling legend items on click events seems to reorder the legend items.

 it occurs randomly, the order it reorders to is always the same, and the hide/show/toggle functionality is not broken, only the order of the items are rearranged.

**Comment by Codebreaker101** [comment]
In addition to this, toggling legend after updating data after the order of the legend has been changed results in the data it toggles to be switched as well.

Here's a fiddle to demonstrate: 
http://jsfiddle.net/7xsrnhsf/7/

Steps to recreate:
1. Go to fiddle
2. Toggle off, then on one of the legends data set.
3. Click on the button **test**
4. Hover/toggle one of the legend data set

As can be seen, when hovering, the wrong data set is showed. Also, when toggling data the wrong data is turned off. 

After some testing the bug seems to be related to the grouping of the data. When commenting out the group option, the data is shown correctly when toggling.

**Comment by Codebreaker101** [comment]
So, the problem is that, seemingly out of nowhere variable that holds graph data (**$$.data.targets**) gets reordered. But only when grouping is used. And then only sometimes. 
@Pixcalibur on your example did you use grouping?

**Comment by Pixcalibur** [comment]
Yes I did, also the problem does not seem to happen in 0.4.10 version of c3 only 0.4.11

**Comment by Codebreaker101** [comment]
https://github.com/c3js/c3/pull/1741/files 

This would be temporary fix. For me it reorders the data (bottom blue bar is now shown on top of orange bar) but for now for me it doesn't matter. Ill try to work on a fix as soon as I can.

**Comment by schwassj** [comment]
alternatively pass data.order = null, provided you've ordered your data as you pass it to c3.

**Comment by Dorfnox** [comment]
Was this fixed? It's still bugged for me.
